genbindings.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
```

Comment:
I just started going through this file but intuitively it seems like it could use a few more comments throughout